JASP-Engine/JASP/R/commonerrorcheck.R

```diff
@@ -542,10 +577,24 @@
     dataset <- split(dataset, groupingData)
   }
   for (d in seq_along(dataset)) {
-    cormat <- do.call(corFun, c(list(dataset[[d]]), corArgs))
+    
+    if (is.null(options$wlsWeights)) {
+      options$wlsWeights <- ""
+    }
+    
+    if (! options$wlsWeights == "") {
```

Comment:
by directly supplying weights in the argument, this can become:

    if (!is.null(weights)) {
      cov <- cov.wt(dataset[[d]], weights)
      ...
    } else {
      cormat <- do.call(corFun, c(list(dataset[[d]]), corArgs))
    }
    err <- .checkVarCovMatrix(cormat, nrow = FALSE, symm = FALSE)

that way, the check can still be used even if people don't store the weights in `options$wlsWeights`.